src/test/scala/io/iohk/ethereum/vm/DataWordSpec.scala

```diff
@@ -114,6 +168,57 @@ class DataWordSpec extends FunSuite with PropertyChecks {
     }
   }
 
+  test("signExtend") {
+    assert(DataWord(42).signExtend(DataWord(3)) == DataWord(42))
+    assert(DataWord(42).signExtend(DataWord(1)) == DataWord(42))
+    assert(DataWord(42).signExtend(DataWord(-1)) == DataWord(42))
+    assert(DataWord(42).signExtend(DataWord(0)) == DataWord(42))
+    assert(DataWord(42).signExtend(DataWord(Size)) == DataWord(42))
+    assert(DataWord(42).signExtend(DataWord(Size + 1)) == DataWord(42))
+    assert(DataWord(-42).signExtend(DataWord(Size)) == DataWord(-42))
+    assert(DataWord(-42).signExtend(DataWord(Size + 1)) == DataWord(-42))
+    assert(DataWord(-42).signExtend(DataWord(-11)) == DataWord(-42))
+    assert(DataWord(-1).signExtend(DataWord(1)) == DataWord(-1))
+    assert(DataWord(-1).signExtend(DataWord(1)) == DataWord(-1))
+    // 1737215.signExtend(1) == 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff81ff
```

Comment:
Also we could use more numbers like that, with 42 the result is always either 42 or 0.